### ISSUE TITLE
Questions about WADI dataset

### ISSUE THREAD

**Issue by ly1112** [d-ailin/GDN#17]
Hi, thanks so much for your sharing.

I am now trying to reproduce your results on WADI dataset.

Now I use the following hyperparameters as below:

![微信图片_20220119235150](https://user-images.githubusercontent.com/40157155/150166497-370fd4d9-e0cd-4854-86d4-a74a594e5df1.jpg)

But the results are around 4% below your results in the paper.

![image](https://user-images.githubusercontent.com/40157155/150166610-2af0b502-5bdc-4381-890c-c387b9a42263.png)

May I ask do you have other hyper-parameter to set or there are something I need to change? And I am using the OCT data to train and test( If I use NOV data to train the results are even lower.) I actually achieved the same results on the SWAT with the setting you mentioned in another issue but the WADI seems to be confused. Thanks a lot and looking forward to your reply!

**Comment by d-ailin** [comment]
1. I think the overall setting is fine, and maybe you could try val_ratio with 0.02 as well.
2. For the dataset, we used the first version of the WADI dataset where the training data is from Sep to Oct and the test data is in Oct. 
3. Though we have fixed the random seeds, the trained result will still vary. So you could also make several trials to see.

**Comment by ly1112** [comment]
> 1. I think the overall setting is fine, and maybe you could try val_ratio with 0.02 as well.
> 2. For the dataset, we used the first version of the WADI dataset where the training data is from Sep to Oct and the test data is in Oct.
> 3. Though we have fixed the random seeds, the trained result will still vary. So you could also make several trials to see.

Thanks so much for your answer and do you mean I can try different randomm seeds to achieve your results? Thanks a lot!

**Comment by d-ailin** [comment]
I mean that even fixing the random seeds like 2 in our program, rerunning the program can still achieve varied results (This might be due to the PyTorch implementation issues, they have mentioned it in their webpages). So that You could just fix the current seed to rerun it several times to see.

**Comment by ly1112** [comment]
> I mean that even fixing the random seeds like 2 in our program, rerunning the program can still achieve varied results (This might be due to the PyTorch implementation issues, they have mentioned it in their webpages). So that You could just fix the current seed to rerun it several times to see.

Thanks so much for your suggestions. I am also interested in anomaly detection, may I ask can you leave an email address so I can directly send it to you when I have some doubt about this? Thanks a lot!

**Comment by d-ailin** [comment]
Sure, you could contact me via ailin@u.nus.edu. You could feel free to close this issue or leave a comment if you have any other problems. Thanks.

**Comment by ly1112** [comment]
> Sure, you could contact me via [ailin@u.nus.edu](mailto:ailin@u.nus.edu). You could feel free to close this issue or leave a comment if you have any other problems. Thanks.

Thanks a lot for your kind help!

**Comment by peerschuett** [comment]
Hi @ly1112, did you successfully reproduce the results on the WADI dataset? 
I am trying to do the same and running into bad results as well.
